R/utils.R

```diff
@@ -31,3 +31,16 @@ apply_plot_modifications <- function(plot_obj, plot_type, dblclicking, ranges) {
     plot_obj
   }
 }
+
+#' This function opens a PDF graphics device using \code{\link[grDevices]{pdf}} to suppress
+#' the plot display in the IDE. The purpose of this function is to avoid opening graphic devices
+#' directly in the IDE.
+#'
+#' @param x lazy binding which generates the plot(s)
+#'
+#' @keywords internal
+plot_suppress <- function(x) {
+  grDevices::pdf(nullfile())
+  on.exit(grDevices::dev.off())
+  force(x)
+}
```

Comment:
I've defined `plot_suppress()` as an internal function here because `teal.widget `doesn't rely on teal.code. Adding this function would introduce a dependency, and I wanted to avoid creating unnecessary dependencies for such a small function.

Reply:
Oh, I did not check for this. I'm surprised how `teal.widgets` is completely isolated from any other teal package. In that case, I think it's okay to have this. I can't think of a better way to reduce DRY.

Reply:
It's not so much isolated as it is the "bottom dependency".